staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -92,24 +100,33 @@ func parseBool(key string) bool {
 	return value
 }
 
-// ConverterImpl knows how to convert between interface{} and
+// unstructuredConverter knows how to convert between interface{} and
 // Unstructured in both ways.
-type converterImpl struct {
+type unstructuredConverter struct {
 	// If true, we will be additionally running conversion via json
 	// to ensure that the result is true.
 	// This is supposed to be set only in tests.
 	mismatchDetection bool
+	// comparison is the default test logic used to compare
+	comparison conversion.Equalities
 }
 
-func NewConverter(mismatchDetection bool) Converter {
-	return &converterImpl{
+// NewUnstructuredConverter creates an UnstructuredConverter that accepts JSON typed maps and translates them
+// to Go types via reflection. If mismatchDetection is true, the converter will crash and exit if
+// the reflection path does not match a decode and encode to JSON for that type.
+func NewUnstructuredConverter(mismatchDetection bool, comparison ...conversion.Equalities) UnstructuredConverter {
```

Comment:
doc vararg behavior... if you only honor 0 and 1, fail if > 1 is passed

Reply:
actually, separate constructor for special snowflakes that have their own ideas about equality would be clearer